common/src/main/java/net/adoptopenjdk/icedteaweb/jvm/JvmUtils.java

```diff
@@ -54,71 +72,188 @@ private static boolean isValidStartingArgument(final String argument) {
+                "-XX:NewRatio", /* set Ratio of new/old gen sizes */
+                "-XX:NewSize", /* set initial size of new generation */
+                "-XX:MaxNewSize", /* set max size of new generation */
+                "-XX:PermSize", /* set initial size of permanent gen */
+                "-XX:MaxPermSize", /* set max size of permanent gen */
+                "-XX:MaxHeapFreeRatio", /* heap free percentage (default 70) */
+                "-XX:MinHeapFreeRatio", /* heap free percentage (default 40) */
+                "-XX:UseSerialGC", /* use serial garbage collection */
+                "-XX:-UseSerialGC",
+                "-XX:ThreadStackSize", /* thread stack size (in KB) */
+                "-XX:MaxInlineSize", /* set max num of bytecodes to inline */
+                "-XX:ReservedCodeCacheSize", /* Reserved code cache size (bytes) */
+                "-XX:MaxDirectMemorySize", /* set maximum direct memory size */
+                "-XX:PrintCMSStatistics", /* If > 0, Print statistics about the concurrent collections */
+                "-XX:SurvivorRatio",                    /* Sets the ratio of the survivor spaces */
+                "-XX:MaxTenuringThreshold",             /* Determines how much the objects may age */
+                "-XX:CMSMarkStackSize",
+                "-XX:CMSMarkStackSizeMax",
+                "-XX:CMSIncrementalDutyCycleMin",       /* The percentage which is the lower bound on the duty cycle */
+                "-XX:ParallelCMSThreads",
+                "-XX:ParallelGCThreads",                /* Sets the number of parallel GC threads */
+                "-XX:CMSInitiatingOccupancyFraction",   /* Sets the threshold percentage of the used heap */
+                "-XX:+UseCompressedOops",               /* Enables compressed references in 64-bit JVMs */
+                "-XX:GCPauseIntervalMillis",
+                "-XX:MaxGCPauseMillis",                 /* A hint to the virtual machine to pause times */
+                "-XX:+CMSIncrementalMode",              /* Enables the incremental mode */
+                "-XX:MaxMetaspaceSize",                 /* Sets an upper limit on memory used for class metadata */
+                "-XX:StringDeduplicationAgeThreshold",
+                "-XX:GCTimeLimit",
+                "-XX:GCHeapFreeLimit"
+        };
+    }
+
+    /**
+     * https://docs.oracle.com/javase/8/docs/technotes/guides/javaws/developersguide/syntax.html#secure-property
+     * <p>
+     * Properties set in the JNLP file will normally be set by Web Start after the VM is started but before
+     * the application is invoked. These properties are considered "secure" properties and can be passed as
+     * -Dkey=value arguments on the java invocation command line. The following properties are predefined secure
+     * properties and will be passed to the VM in this way.
+     *
+     * @return
+     */
+    private static String[] getValidSecureProperties() {
+        return new String[]{
```

Comment:
Could you please group the properties by the prefix.
Such that all `sun.` and `http.` are together. This would improve readability and allow faster lookup in the future

Reply:
The properties have been taken from Java 8 doc. New props in Java 8 follow those in Java 6. It was easier to compare properties this way against the doc.